core/src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala

```diff
@@ -88,6 +92,8 @@ trait CassandraRecovery extends CassandraTagRecovery with TaggedPreparedStatemen
           someReadConsistency,
           someReadRetryPolicy,
           extractor = Extractors.persistentRepr(eventDeserializer, serialization))
+        // run the query on the journal dispatcher (not the queries dispatcher)
+        .withAttributes(ActorAttributes.dispatcher(sessionSettings.pluginDispatcher))
```

Comment:
We have to be sure that this isn't just adding an outer asynchronous boundary. Might be more clear and safe to pass in the dispatcher as parameter to `queries. eventsByPersistenceId`?